Lib/test/_test_multiprocessing.py

```diff
@@ -3149,6 +3149,29 @@ def test_rapid_restart(self):
             if hasattr(manager, "shutdown"):
                 self.addCleanup(manager.shutdown)
 
+#
+# Issue 106558: Manager exceptions should not create cyclic references.
+#
+
+class TestManagerExceptions(unittest.TestCase):
+    def setUp(self):
+        self.mgr = multiprocessing.Manager()
+
+    def tearDown(self):
+        self.mgr.shutdown()
+        self.mgr.join()
+
+    def test_queue_get(self):
+        queue = self.mgr.Queue()
+        if gc.isenabled():
+            gc.disable()
+            self.addCleanup(gc.enable)
+        try:
+            queue.get_nowait()
+        except pyqueue.Empty as e:
+            wr = weakref.ref(e)
+        self.assertEqual(wr(), None)
```

Comment:
Actually, this doesn't cover the changes to lines 93-96.  Can we add a test for that?

Reply:
Yep, added, though now I'm having second thoughts:

Apparently `dispatch` as the standard library actually uses it will only ever* respond with `kind` being `'#RETURN'` (then `convert_to_error` is not called) or `'#TRACEBACK'` (then `result` is a string so it's not a link in a reference cycle).

The most I could imagine that changing is if someone tries to introduce a custom connection class, so the test vaguely gestures in that direction. But since the caller of `dispatch` has to know about this `(kind, result)` return type, it still feels like exposing internal implementation details to the test. Now I'm starting to wonder if the better tradeoff would be to revert lines 93-96—maybe at most leaving some version of this comment in the commit message to explain why we're only wrapping one of the two calls to `convert_to_error`.

(*Proof: All calls to `dispatch` are on freshly created `self._Client` instances, so they all are answered by `serve_forever → accepter → handle_request → _handle_request`, and `_handle_request` only responds with those two `kind`s.

For context, the one line that does generate `'#ERROR'` is inside `serve_client`—only contactable by reusing a connection on which we previously `dispatch`ed `accept_connection`, and the one place that happens is when `_callmethod` calls `_connect`. Even the `dispatch` call that's later in `_callmethod` comes after replacing `conn` with a fresh `self._Client`.)

Reply:
I would leave the fix in, I think this test is fine, good enough to prevent regression if someone reverts the change to the code (which I think we should have).